skpro/distributions/hurdle.py

```diff
@@ -0,0 +1,198 @@
+"""Hurdle distribution implementation."""
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+from skpro.distributions.base import BaseDistribution
+from skpro.distributions.left_truncated import LeftTruncated
+
+
+# TODO: how to handle index/columns in these transformed distributions? must they be
+#  the same as the original distribution?
+class Hurdle(BaseDistribution):
+    r"""A Hurdle distribution.
+
+    Combines a Bernoulli gate for zero vs. non-zero outcomes with a zero-truncated
+    distribution for the positive outcomes. Mathematically this can be expressed as:
+
+    .. math::
+        Y_t = \begin{cases}
+                X \sim f(x \vert x > 0) &\text{ with probability } \pi, \\
+                0 &\text{ with probability } 1 - \pi,
+              \end{cases}
+
+    where :math:`\pi` is the probability of getting a non-zero value, and
+    :math:`f(x \vert x > 0)` is the probability mass function of the zero-truncated
+    distribution.
+
+    Parameters
+    ----------
+    p : np.ndarray
+        The probability of getting a non-zero value.
+
+    distribution : LeftTruncated
+        The zero-truncated distribution for positive outcomes.
+
+    """
+
+    _tags = {
+        "capabilities:approx": ["energy", "cdf"],
+        "capabilities:exact": ["ppf", "mean", "var", "log_pmf", "pmf"],
+        "distr:measuretype": "discrete",
+        "distr:paramtype": "parametric",
+        "broadcast_init": "on",
+    }
+
+    def __init__(
+        self,
+        p: ArrayLike,
+        distribution: LeftTruncated,
+        index=None,
+        columns=None,
+    ):
+        assert (
+            distribution.lower == 0
```

Comment:
this is a contract not all distributions satisfy - if `distribution` does not have the `lower` attribute, this will fail.

What I would suggest instead: check if the distribution is `LeftTruncated`, if not, left truncate it at zero.

A more idiomatic approach would be to add supports that are inspectable to all distributions, though that is a major change and out of scope (I wonder if @vascosch92 is stlil working on this?)